cli/internal/ffi/ffi.go

```diff
@@ -25,7 +30,12 @@ func Unmarshal[M proto.Message](b C.Buffer, c M) error {
 		return err
 	}
 
-	b.Free()
+	// free the buffer on the rust side
+	//
+	// SAFETY: do not use `C.free_buffer` to free a buffer that has been allocated
+	// on the go side. If you happen to accidentally use the wrong one, you can
+	// expect a segfault on some platforms. This is the only valid callsite.
+	C.free_buffer(b)
```

Comment:
I am not certain that this is safe. `toBytes` returns a Go-owned slice. Can you document what happens when Go goes to free it?